pkgs/build-support/buildenv/builder.pl

```diff
@@ -122,14 +122,15 @@ sub findFiles {
     }
 
     unless (-d $target && ($oldTarget eq "" || -d $oldTarget)) {
-        if ($ignoreCollisions) {
-            warn "collision between `$target' and `$oldTarget'\n" if $ignoreCollisions == 1;
-            return;
-        } elsif ($checkCollisionContents && checkCollision($oldTarget, $target)) {
-            return;
-        } else {
-            die "collision between `$target' and `$oldTarget'\n";
+        if (!$checkCollisionContents || !checkCollision($oldTarget, $target)) {
+            my $collisionMessage = "collision between `$target' and `$oldTarget'\n";
+            if ($ignoreCollisions) {
+                warn $collisionMessage;
+            } else {
+                die $collisionMessage;
+            }
```

Comment:
it could be shorter:

```diff
-            if ($ignoreCollisions) {
-                warn $collisionMessage;
-            } else {
-                die $collisionMessage;
-            }
+            die $collisionMessage unless $ignoreCollisions;
+            warn $collisionMessage;
```

Reply:
Updated.